### PR TITLE
Manual page

### PR DESCRIPTION
**Pull request by rrobgill**
**By submitting this pull request, I confirm the following:** 
*please fill any appropriate checkboxes, e.g: [X]*

- [X] I have read and understood the [contributors guide](https://github.com/pi-hole/pi-hole/blob/master/CONTRIBUTING.md), as well as this entire template.
- [X] I have made only one major change in my proposed changes.
- [X] I have commented my proposed changes within the code.
- [X] I have tested my proposed changes, and have included unit tests where possible.
- [X] I am willing to help maintain this change if there are issues with it later.
- [X] I give this submission freely and claim no ownership.
- [X] It is compatible with the [EUPL 1.2 license](https://opensource.org/licenses/EUPL-1.1)
- [X] I have squashed any insignificant commits. ([`git rebase`](http://gitready.com/advanced/2009/02/10/squashing-commits-with-rebase.html))

Please make sure you [Sign Off](https://github.com/pi-hole/pi-hole/wiki/How-to-signoff-your-commits.) all commits. Pi-hole enforces the [DCO](https://github.com/pi-hole/pi-hole/wiki/Contributing-to-the-project).

---
**What does this PR aim to accomplish?:**
Adds a conventional linux manual page for the pihole command.

**How does this PR accomplish the above?:**

Adds code to install.sh to detect if man pages are installed on the system, and if so installs a man page and updates man page database. Makes no changes to lightweight systems without man pages installed.

Adds code to uninstall.sh to detect if a pihole manpage was installed, and if so removes it and updates the man page database. 

Adds pihole.8 a conventional linux man page for pihole.

**What documentation changes (if any) are needed to support this PR?:**

...it is a documentation change. 

Any future alteration to the command-line paramaters of the pihole script would need to be reflected in changes to pihole.8

---

### CODE REVIEW

File: automated install/basic-install.sh [hunk 1]
```diff
@@ -1193,6 +1193,37 @@ installConfigs() {
...
+  cp ${PI_HOLE_LOCAL_REPO}/pihole.8 /usr/local/share/man/man8/pihole.8
+  if mandb -q &>/dev/null; then
+    # Updated successfully
+    echo -e "${OVER}  ${TICK} man page installed and database updated"
+    return
+  else
+    # Something is wrong with the system's man installation, clean up 
+    # our file, (leave everything how we found it).
+    rm /usr/local/share/man/man8/pihole.8
+    echo -e "${OVER}  ${INFO} man page db not updated, man page not installed"
```
Comment (Mcat12) [inline automated install/basic-install.sh]: This can be `CROSS`

File: automated install/basic-install.sh [hunk 2]
```diff
@@ -1193,6 +1193,37 @@ installConfigs() {
   fi
 }
 
+install_manpage() {
+  # Copy Pi-hole man page and call mandb to update man page database
+  # Default location for man files for /usr/local/bin is /usr/local/share/man
+  # on lightweight systems may not be present, so check before copying.
+  echo -en "  ${INFO} Testing man page installation"
+  if ! command -v mandb; then
```
Comment (Mcat12) [inline automated install/basic-install.sh]: Add `&> /dev/null`:
```
  [i] Testing man page installation/usr/bin/mandb
```

File: pihole.8 [hunk 3]
```diff
@@ -0,0 +1,326 @@
+.TH "Pi-hole" "8" "Pi-hole" "Pi-hole" "May 2018"
+.SH "NAME"
+
+Pi-hole : A black-hole for internet advertisements
+.br
+.SH "SYNOPSIS"
```
Comment (Mcat12) [inline pihole.8]: Missing `pihole -v`

File: pihole.8 [hunk 4]
```diff
@@ -0,0 +1,326 @@
...
+.br
+	Reconfigure or Repair Pi-hole subsystems
+.br
+
+\fB-t, tail\fR
+.br
+	View the live output of the Pi-hole log
+.br
+
+\fB-a, admin\fR [options] Console
```
Comment (Mcat12) [inline pihole.8]: Is `Console` supposed to be here?

File: pihole.8 [hunk 5]
```diff
@@ -0,0 +1,326 @@
...
+\fB-up, updat\fBe\fR\fR\fBPihole\fR [--check-only]	
+.br
+	Update Pi-hole subsystems 
+.br
+		Add '--check-only' to exit script before update is performed.
+.br
+
+\fB-v, version\fR [repo] [options]	
+.br
+	Show installed versions of Pi-hole, Admin Console &amp; FTL
```
Comment (Mcat12) [inline pihole.8]: `&amp;` shows up as itself, not `&`

File: pihole.8 [hunk 6]
```diff
@@ -0,0 +1,326 @@
...
+	Specify whether the Pi-hole log should be used 
+.br
+
+	(Logging options):
+.br
+		on Enable the Pi-hole log at /var/log/pihole.log
+.br
+		off Disable and flush the Pi-hole log at /var/log/pihole.log
+.br
+		off noflush Disable the Pi-hole log at /var/log/pihole.log
```
Comment (Mcat12) [inline pihole.8]: These should have their descriptions indented

File: pihole.8 [hunk 7]
```diff
@@ -0,0 +1,326 @@
...
+.br
+		#m  Disable Pi-hole functionality for # minute(s)
+.br
+
+\fBrestartdns\fR
+.br
+	Restart Pi-hole subsystems
+.br
+
+\fBcheckout\fR [repo][branch]
```
Comment (Mcat12) [inline pihole.8]: Missing a space between repo and branch

File: pihole.8 [hunk 8]
```diff
@@ -0,0 +1,326 @@
...
+		web         Change the branch of Admin Console subsystem
+.br
+		ftl         Change the branch of Pi-hole's FTL subsystem
+.br
+	(branch options):
+.br
+		master      Update subsystems to the latest stable release
+.br
+		dev         Update subsystems to the latest development release
+.br
```
Comment (Mcat12) [inline pihole.8]: You can specify a branch name as well

File: pihole.8 [hunk 9]
```diff
@@ -0,0 +1,326 @@
...
+	
+.br
+	Switching Pi-hole subsystem branches
+.br
+	
+.br
+	\fBpihole checkout master\fR			Switch to master branch
+.br
+	\fBpihole checkout core dev\fR		Switch to core development branch
```
Comment (Mcat12) [inline pihole.8]: Some of these descriptions do not line up with the rest

File: pihole.8 [hunk 10]
```diff
@@ -0,0 +1,326 @@
...
+.br
+
+	Whitelist/blacklist manipulation
+.br
+
+	\fBpihole -w iloveads.example.com\fR	Add "iloveads.example.com" to whitelist
+.br
+	\fBpihole -b -d noads.example.com\fR	Remove "noads.example.com" from blacklist 
+.br
+	\fBpihole -wild ads.example\fR		Add "ads.example" as wildcard - would block ads.example.net
```
Comment (Mcat12) [inline pihole.8]: This is backwards. A better example would be:
> Add "example.net" as wilcard - would block ads.example.net

File: pihole.8 [hunk 11]
```diff
@@ -0,0 +1,326 @@
...
+	
+.br
+	\fBpihole -g\fR						Update the list of ad-serving domains
+.br
+
+	Displaying version information
+.br
+	
+.br
+	\fBpihole -v -c\fR					Display the current version of AdminLTE
```
Comment (Mcat12) [inline pihole.8]: This would display current versions of all three components

File: pihole.8 [hunk 12]
```diff
@@ -0,0 +1,326 @@
...
+.br
+	
+.br
+	\fBpihole checkout master\fR			Switch to master branch
+.br
+	\fBpihole checkout core dev\fR		Switch to core development branch
+.br
+.SH "SEE ALSO"
+
+dmasq(8), lighttpd(8)
```
Comment (Mcat12) [inline pihole.8]: Misspelling of "dnsmasq"

File: pihole.8 (hunk 3 above)
Comment (rrobgill) [reply pihole.8]: fixed

File: pihole.8 [hunk 13]
```diff
@@ -56,269 +58,273 @@ Available commands and options:
...
 .br
 
-	\fBpihole -w iloveads.example.com\fR	Add "iloveads.example.com" to whitelist
+    \fBpihole -w iloveads.example.com\fR  Add "iloveads.example.com" to whitelist
 .br
-	\fBpihole -b -d noads.example.com\fR	Remove "noads.example.com" from blacklist 
+    \fBpihole -b -d noads.example.com\fR  Remove "noads.example.com" from blacklist 
 .br
-	\fBpihole -wild ads.example\fR		Add "ads.example" as wildcard - would block ads.example.net
+    \fBpihole -wild example\fR            Add "example" as wildcard - would block ads.example.net, example.com etc.
```
Comment (Mcat12) [inline pihole.8]: The wildcard block only blocks subdomains, so `example` would not block `example.com`, but `example.com` would block `test.example.com`.

File: pihole.8 (hunk 7 above)
Comment (Mcat12) [reply pihole.8]: Still missing a space.

**Pull request merged**